WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelper.kt

```diff
@@ -437,6 +438,12 @@ class PostListItemUiStateHelper @Inject constructor(
         return buttonTypes
     }
 
+    private fun MutableList<PostListButtonType>.addIf(condition: Boolean, buttonType: PostListButtonType) {
```

Comment:
@antonis I was thinking of using this extension for all buttons which are under a simple condition captured by a boolean variable.
WDYT?

Reply:
Great work leveraging our [Coding style tools ](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/docs/coding-style.md) and detecting the `LongMethod` issue 👍 
Though I'm a big fan of kotlin extensions I'm a bit skeptical on this since it applies just on the specific usecase/method and is not a generic extension that is discoverable and can be used in other parts of the project 🤔 
What are your thoughts on extracting some of the functionality of the `createButtonTypes` method to private functions to reduce the complexity. For example we could extract[ the last `when` statement](https://github.com/wordpress-mobile/WordPress-Android/pull/15570/files#diff-2238863dfadc135bc1b6359a21a855541d2dcf3d8cd2742d2fa91a81ea03e701L429) that adds the proper delete/trash button. Wdyt?

Reply:
Thank you, that was indeed the constraint here!

I can resonate with your skepticism as I also don't see it as something discoverable and reusable in other parts of the code, it has a pretty narrow scope.

Already did some thinking while working on the task and also saw this suggestion (to extract the last when statement for the proper delete/trash button) as a possible improvement!

I'm going to explore this suggestion and which other functionalities can be extracted while still maintaining a good overview of the buttons that are added!

Reply:
Updated this too!
Thank you for the suggestions

Reply:
Thank you for tackling this @ovitrif 🙇 
The changes introduced with 8fac44dd0f21a840733f3790fd5029423ad75a5a, 6e3550efebf5c1f6fad6bddf3fca9f86a191b1c0, 6551903b6666d97db9ba1e718ae759325340c4a9 and 48b7432792a3a4c7ff6a717d6002b0456c1266ad LGTM 🎉